antarest/study/business/allocation_management.py

```diff
@@ -0,0 +1,95 @@
+from typing import List
+
+from pydantic import BaseModel
+from starlette.responses import JSONResponse
+
+from antarest.study.business.utils import (
+    FormFieldsBaseModel,
+    execute_or_add_commands,
+)
+from antarest.study.model import Study
+from antarest.study.storage.storage_service import StudyStorageService
+from antarest.study.storage.variantstudy.model.command.update_config import (
+    UpdateConfig,
+)
+
+
+class AllocationField(FormFieldsBaseModel):
+    area_id: str
+    coefficient: float
+
+
+class AllocationFormFields(BaseModel):
+    allocation: List[AllocationField]
+
+
+class AllocationManager:
+    """
+    Provides functionality for getting and updating allocation fields for a given area
+    """
+
+    def __init__(self, storage_service: StudyStorageService) -> None:
+        self.storage_service = storage_service
+
+    def get_field_values(
+        self, study: Study, area_id: str, all_areas
+    ) -> JSONResponse:
+        file_study = self.storage_service.get_storage(study).get_raw(study)
+        allocation_data = file_study.tree.get(
+            f"input/hydro/allocation/{area_id}".split("/"), 2
+        )
+
+        areas_ids = [area.id for area in all_areas]
+
+        if not allocation_data:
+            error_msg = f"No allocation data found for area {area_id}"
```

Comment:
should raise an exception.

Create an exception or pick-up one in [exceptions.py](https://github.com/AntaresSimulatorTeam/AntaREST/blob/7647253d6f2859ad6e50923d874ac3131ef2da9b/antarest/core/exceptions.py)